zcash_keys/src/address.rs

```diff
@@ -293,20 +339,24 @@ mod tests {
             Some(fvk.address_at(0u32, orchard::keys::Scope::External))
         };
 
+        #[cfg(feature = "sapling")]
         let sapling = {
             let extsk = sapling::spending_key(&[0; 32], 0, AccountId::ZERO);
             let dfvk = extsk.to_diversifiable_full_viewing_key();
             Some(dfvk.default_address().1)
         };
 
-        let transparent = { None };
+        let transparent = None;
 
-        #[cfg(feature = "orchard")]
+        #[cfg(all(feature = "orchard", feature = "sapling"))]
         let ua = UnifiedAddress::from_receivers(orchard, sapling, transparent).unwrap();
 
-        #[cfg(not(feature = "orchard"))]
+        #[cfg(all(not(feature = "orchard"), feature = "sapling"))]
         let ua = UnifiedAddress::from_receivers(sapling, transparent).unwrap();
 
+        #[cfg(all(not(feature = "orchard"), not(feature = "sapling")))]
+        let ua = UnifiedAddress::from_receivers(transparent).unwrap();
```

Comment:
This test case should be failing at the unwrap. I suspect it doesn't fail CI because we don't (and can't really) test all feature flag combinations. Move it into a separate test (or early-return after checking you get `None`).